docs/DeveloperGuide.md

```diff
@@ -140,6 +140,54 @@ Classes used by multiple components are in the `seedu.clinic.commons` package.
+The feature also allows user to delete a product from a specific warehouse or supplier __(case 2)__.
+The deletion is limited to the items shown in the ui, i.e the displayed results, and is done 1 item at a time.
+
+#### How it is implemented
+The `delete` feature is mainly facilitated via the `DeleteCommand` class.
+It extends the abstract `Command` class, with the ability to handle __case 1__ and __case 2__ separately:
+
+* `DeleteCommand#executeWarehouseRelatedDeletion()` — Delete the entire target warehouse or its specific product.
+* `DeleteCommand#executeSupplierRelatedDeletion()` — Delete the entire target supplier or its specific product.
+
+Given below is a description about the mechanism of deletion.
+
+Step 1. CLI-nic's `parser` will parse the user input and if the `delete` command word is present, the parser will try to parse the
+input into a valid `DeleteCommand` via **DeleteCommandParser**.
+As usual, checks for compulsory prefixes and valid arguments (`ct/TYPE` and `i/INDEX` in this case) are done.
+If two entries of `ct/TYPE` are found, the latter entry will be used as the argument.
+If the deletion command asks for deletion of product (indicated by the `TYPE` keywords `ps` and `pw`), the additional check
+for prefix `pd/` and valid product name will also be conducted.
+The code will throw a **ParseException** if the check fails.
+Afterwards, all the valid arguments will form a `DeleteCommand`, which will be executed.
+
+Step 2. The `DeleteCommand` is executed via a call from `LogicManager`. The execution is first classified into Supplier-related deletion
+and Warehouse-related deletions via the `targetType` attribute. Under each category, The execution splits into deletion of an entire supplier/warehouse
+or its particular product.
+
+Step 3. The method will retrieve the displayed list of warehouse/supplier via `model#getFilteredWarehouseList()`.
+It locates the warehouse/supplier user wants to delete (or from whom the product to delete) via the `index` passed in.
+
+<div markdown="span" class="alert alert-info">:information_source:
+**Note:** If the index passed in is larger than the size of the list, an error will be raised and the deletion will terminate.
+</div>
+
+Step 4. If the user wants to delete an entire warehouse/supplier entry, `model#deleteWarehouse` will remove the entry
+from the list in the `model`.
+
+If the user wants to delete a product inside the entry, the set of product for the warehouse/supplier entry will be retrieved first.
+The `warehouse#getProductByName` will give the target product to delete from the product name parsed, and the retrieved product set will
+remove this product from the set. Afterwards, the updated product set will replace the old set in the warehouse. The model will also 
+update the warehouse with the new warehouse with the target product deleted.
+
+Step 5. With the deletion completed, a `CommandResult` will be returned to the `LogicManager` with the success message, which will
```

Comment:
I think it should be " with a success message".

Reply:
I guess I was referring to the particular success message with the delete command. Anyway I think your suggestion looks clearer to readers, thanks!